dashboard/src/main/java/org/apache/uniffle/dashboard/web/proxy/WebProxyServlet.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.dashboard.web.proxy;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.proxy.ProxyServlet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.dashboard.web.JettyServerFront;
+
+public class WebProxyServlet extends ProxyServlet {
+
+  private String hostAndPort;
+  private static final Logger LOG = LoggerFactory.getLogger(JettyServerFront.class);
+
+  public WebProxyServlet(String hostAndPort) {
+    this.hostAndPort = hostAndPort;
+  }
+
+  @Override
+  protected String rewriteTarget(HttpServletRequest clientRequest) {
+    if (!validateDestination(clientRequest.getServerName(), clientRequest.getServerPort())) {
+      return null;
+    }
+    StringBuilder target = new StringBuilder();
+    String targetUri = "http://" + hostAndPort + "/";
```

Comment:
May use https

Reply:
> May use https

None of the servers support https, so that's not necessary.

Reply:
https://github.com/apache/incubator-uniffle/blob/be106976474cc02a614ca842421670debf50f7aa/common/src/main/java/org/apache/uniffle/common/web/JettyServer.java#L83-L85